ui/src/pages/deployment-details/deployment-details-form/section-location/geo-search/geo-search.tsx

```diff
@@ -0,0 +1,42 @@
+import { ComboBox } from 'design-system/components/combo-box/combo-box'
+import { MarkerPosition } from 'design-system/map/types'
+import { useMemo, useState } from 'react'
+import { STRING, translate } from 'utils/language'
+import { useGeoSearch } from './useGeoSearch'
+
+export const GeoSearch = ({
+  onPositionChange,
+}: {
+  onPositionChange: (position: MarkerPosition) => void
+}) => {
+  const [searchString, setSearchString] = useState('')
+  const { data } = useGeoSearch(searchString)
+
+  const items = useMemo(() => {
+    if (!data?.length) {
+      return []
+    }
+    return data.map((result) => ({
+      id: result.osmId,
+      label: result.displayName,
+    }))
+  }, [data])
```

Comment:
What is the reasoning for using useMemo here instead of deriving the data in the render function?

Reply:
Since `items` is an object, if we would not use `useMemo`, the object would get a new ref each render, which might cause unnecessary rendering, especially since we pass it as a prop to child components. Probably not a huge impact in practice though. I know React is pretty optimised to handle re rendering. Might be better to optimise for readability?

Reply:
That is sound reasoning. I was just curious :D